Mathlib/Order/KrullDimension.lean

```diff
@@ -162,6 +164,36 @@ lemma height_mono : Monotone (α := α) height :=
 @[gcongr] protected lemma _root_.GCongr.height_le_height (a b : α) (hab : a ≤ b) :
     height a ≤ height b := height_mono hab
 
+private lemma height_add_const (a : α) (n : ℕ∞) :
+    height a + n = ⨆ (p : LTSeries α ) (_ : p.last = a), p.length + n := by
+  have hne : Nonempty { p : LTSeries α // p.last = a } := ⟨RelSeries.singleton _ a, rfl⟩
+  rw [height_eq_iSup_last_eq, iSup_subtype', iSup_subtype']
+  rw [Monotone.map_ciSup_of_continuousAt (f := (· + n))
+    (continuousAt_id.add continuousAt_const) (monotone_id.add monotone_const)]
```

Comment:
Horror! Don't use topology for such simple statement :-)

See how [`ENNReal.iSup_add`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Instances/ENNReal.html#ENNReal.iSup_add) is proved and try doing the same for `ENat`

Reply:
That looks very similar to me:
```
theorem iSup_add {ι : Sort*} {s : ι → ℝ≥0∞} [Nonempty ι] : iSup s + a = ⨆ b, s b + a :=
  Monotone.map_ciSup_of_continuousAt (continuousAt_id.add continuousAt_const) <|
    monotone_id.add monotone_const
```

Reply:
I have a non-topological ugly proof in https://github.com/leanprover-community/mathlib4/pull/15344/files, but [was advised](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Topology.20of.20.60ENat.60/near/455447883) to use https://github.com/leanprover-community/mathlib4/pull/15380. Was that ill-advised?

Reply:
Oh, that looks quite bad. Certainly, the elementary theory of heights should not require topology. I will try to open a PR downgrading the imports to `ENNReal.iSup_add`.

Reply:
Thanks for the heads up, and no rush!

Reply:
The PRs are #16735 and #16742. Do you want me to prove the `ENat` versions as well or are you happy to take it from there?

Reply:
I can give it a shot when I get to it, but not sure when that will be, and will probably require some hand holding. So if you have pages it all in and have a clear picture where to go, maybe better go ahead?

Reply:
Okay, will do it myself